### PR TITLE
Dev/android/home page

### PR DESCRIPTION
**Pull request by AlicanM**
Requesting to merge dev/android/home-page into develop.

In this branch,

- the implementation of the Home Page is fully completed with the back-end integrations.
- models are arranged according to the endpoints provided by the back-end team.
- the UI design is completed although it is always open for further development.

For more information about the development process of this task, please check the tracking issue https://github.com/bounswe/bounswe2022group7/issues/351.

### CODE REVIEW

File: android/lib/models/art_item_info_model.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    required this.description,
+    this.category,
+    this.imageUrl,
+    this.labels,
+  });
+
+  factory ArtItemInfo.fromJson(Map<String, dynamic> json) {
+    return ArtItemInfo(
+      id: json['id'],
+      name: json['name'] ?? "",
```
Comment (canatakan) [inline android/lib/models/art_item_info_model.dart:25]: Creating an ArtItemInfo without a name is not a good solution.

File: android/lib/models/art_item_model.dart [hunk 2]
```diff
@@ -1,15 +1,77 @@
...
+    this.lastPrice,
+    required this.commentList,
+    required this.bookmarkedBy,
+  });
+
+  factory ArtItem.fromJson(Map<String, dynamic> json) {
+    return ArtItem(
+      id: json['id'],
+
+      artItemInfo: json['artItemInfo'] != null
```
Comment (canatakan) [inline android/lib/models/art_item_model.dart:33]: In order to escape from null type checks, following this approach is not the correct way in my opinion.

File: android/lib/models/art_item_model.dart [hunk 3]
```diff
@@ -1,15 +1,77 @@
...
+          ? ArtItemInfo.fromJson(json['artItemInfo'])
+          : ArtItemInfo(
+              id: 0,
+              name: "",
+              description: "",
+              category: "",
+              imageUrl: 'https://api.multiavatar.com/Robo.png',
+              labels: List<String>.empty()),
+
+      creator: json['creator'] != null
```
Comment (canatakan) [inline android/lib/models/art_item_model.dart:43]: A similar comment can be made for all of these updates.

File: android/lib/models/post_model.dart [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    required this.type,
+    required this.id,
+    required this.creator,
+    required this.title,
+    required this.description,
+    required this.imageUrl,
+  });
+
+  Widget pageRoute() {
+    return EventPage(id: id);
```
Comment (canatakan) [inline android/lib/models/post_model.dart:24]: I am not sure if this Post page is ever going to redirect to an Art Item Page (or actually even render them).

File: android/lib/models/post_model.dart [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        const Spacer(),
+        IconButton(
+            onPressed: () {},
+            icon: const Icon(Icons.bookmark_add_outlined,
+                color: Colors.black, size: 30.0)),
+      ],
+    );
+  }
+
+/*Row(
```
Comment (canatakan) [inline android/lib/models/post_model.dart:64]: Delete commented codes please.

File: android/lib/models/user_model.dart [hunk 6]
```diff
@@ -21,14 +21,15 @@ class User {
...
-      surname: parsedJson["surname"],
-      email: parsedJson["email"],
-      imageUrl: parsedJson["imageUrl"],
-      username: parsedJson["username"],
-      userType: parsedJson["userType"],
-      age: parsedJson["age"],
-      country: parsedJson["country"],
+      name: parsedJson["name"] ?? "",
+      surname: parsedJson["surname"] ?? "",
+      email: parsedJson["email"] ?? "",
```
Comment (canatakan) [inline android/lib/models/user_model.dart:26]: Should not ever create a User without an email.

**Review by canatakan — CHANGES_REQUESTED** [review]
* There are many problems with the updates in the models. Escaping null check like this is not a correct approach.

* **Much more importantly, your code does not show the Art Item elements.** This PR does NOT provide the functionality that it should.

*  Additionally, **there are some conflicts** after changes in `develop`. 

So, it is impossible to merge this until these issues are resolved.

**Comment by atillaturkmen** [comment]
I have solved the conflicts and fixed the issues Atakan mentioned. Now the homepage works as intended. Therefore, I am merging this pull request.

**Pull request merged**